### PR TITLE
Add comparison ROS vs eCAL.

### PR DESCRIPTION
**Pull request by Kerstin-Keller**
**Pull request type**

 

 

- [x] Documentation content changes

A comparison between eCAL / ROS. Please review for correctness!

### CODE REVIEW

File: doc/rst/advanced/ros_comparison.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  eCAL on the other hand transports binary data. Thus, there is not fix coupling between the message serialization (and thus also the message definitions), and the actual transport of this data.
+  This allows eCAL to be used in conjunction with many serialization libraries such as Protobuf, Flatbuffers, Capnproto or MessagePack.
+  Is it necessary that the schema is backwards compatible? Does the user need fast serialization or easy indexing? 
+  Depending on the usecase, the user can choose whatever format fits them best, on a per message basis.
+  
+- **eCAL is a library (+ tools) while ROS is an ecosystem**
+
+  eCAL was build with modularity and ease of integration in mind. When building communication components, they only need to link to the eCAL core library.
+  eCAL tools may be installed in addition.
+  ROS on the other hand comes with a complete ecosystem, bundling C++ libraries, providing its own builsystem.
```
Comment (kratz00) [inline doc/rst/advanced/ros_comparison.rst:56]:
```suggestion
  ROS on the other hand comes with a complete ecosystem, bundling C++ libraries, providing its own build system.
```

File: doc/rst/advanced/ros_comparison.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+How eCAL differs from ROS(2) today
+----------------------------------------
+eCAL and ROS have both evolved over the years. 
+With ROS2 came Windows support, and the ROS RMW concept allows for more flexible transports (there is also an eCAL implementation of a ROS RMW layer).
+However there are two key factors which sets eCAL apart from ROS today:
+
+- **eCAL is message protocol agnostic**
+
+  ROS's IDL system is tied into ROS. And it's still not backwards compatible regarding message evolution.
+  eCAL on the other hand transports binary data. Thus, there is not fix coupling between the message serialization (and thus also the message definitions), and the actual transport of this data.
```
Comment (kratz00) [inline doc/rst/advanced/ros_comparison.rst:47]:
```suggestion
  eCAL on the other hand transports binary data. Thus, there is no coupling between the message serialization (and thus also the message definitions), and the actual transport of this data.
```

**Review by hannemn — APPROVED** [review]

**Comment by FlorianReimold** [comment]
Please wait for my review that I will add soon

**Review by FlorianReimold — APPROVED** [review]

**Review by rex-schilasky — APPROVED** [review]

File: doc/rst/advanced/ros_comparison.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+eCAL was developed by Continental out of the need of a communication middleware solution.
+At that time, ROS1 was not considered a suitable choice for the following reasons
+
+- **Lack of Windows support**
+
+  ROS1 doesn't support the Windows operating system, which was a hard requirement at the time
+
+- **Data throughput not high enough**
+
+  ROS1 communcation is based on UDP only, for both local and distributed communication.
```
[the remark this answers is not in the document]
Comment (FlorianReimold) [reply doc/rst/advanced/ros_comparison.rst]: I corrected that with my commit before 👍

**Pull request merged**